### PR TITLE
feat: configure ban list by JSON config file

### PR DESCRIPTION
**Pull request by hjaurum**
Supports config ban list by `.config.json` instead of hard code.
For Issue #465 

Unit Test:
image

Local Test:
image

### CODE REVIEW

File: server/config/config_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package config
+
+import "testing"
+
+func TestBanList_IsPackageBaned(t *testing.T) {
```
Comment (ije) [inline server/config/config_test.go:5]: nice to see unit testing👍

**Review by ije — CHANGES_REQUESTED** [review]

**Comment by ije** [comment]
@hjaurum great job! i suggest use plain txt or json as the list input to reduce go deps. thanks 🙏

File: go.mod [hunk 2]
```diff
@@ -35,14 +36,28 @@ require (
...
+	github.com/spf13/pflag v1.0.5 // indirect
+	github.com/subosito/gotenv v1.4.1 // indirect
 	go.etcd.io/bbolt v1.3.6 // indirect
 	golang.org/x/crypto v0.4.0 // indirect
 	golang.org/x/net v0.3.0 // indirect
 	golang.org/x/sys v0.3.0 // indirect
 	golang.org/x/text v0.5.0 // indirect
+	gopkg.in/ini.v1 v1.67.0 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
```
[the remark this answers is not in the document]
Comment (hjaurum) [reply go.mod]: Sure, I have replace the YAML with JSON to reduce the deps.

File: server/config/config.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+var cfgs *Configs
+
+type Configs struct {
```
Comment (ije) [inline server/config/config.go:13]: i like the `Config` structure, maybe we should move all the options to here instead of using flag

**Review by ije — APPROVED** [review]

**Comment by ije** [comment]
LGTM, nice job done 👍

**Pull request merged**